### ISSUE TITLE
Consider "Zip" module

### ISSUE THREAD

**Issue by avishnyakov** [statiqdev/Statiq#178]
Just putting a few more ideas around potential "Wyam" workflow.

Alright, Wyam does the job rendering the web site into "Output" folder. But what's next?
At some point, we'd like to pack it up into zip archive and store somewhere as an "artifact", as a part of the CI process / workflow. Sorta, neat, we've for a build, all good, here is your stuff.

The output folder might be a big one with hundreds files and so on, so I can see "Zip" module been useful in some scenarios - exactly to mimic the "artifact" output of the build process.

The input would be mostly the "Wyam output" folder, and the output - the zip archive.

**Comment by daveaglick** [comment]
:+1:

**Comment by miere43** [comment]
That's dumb, you can just do it with external tools without wasting time implementing almost never-used modules, for example, use a batch file:

```
wyam --output Output
D:\soft\7-zip\7z.exe a -t7z mysite.7z Output/*.*
```

And you are done.

**Comment by avishnyakov** [comment]
Correct, it can be seen as "almost never used".

Most-likely that would be exactly the case unless we run Wyam under TFS/TFSOnline and AppVeyour automated builds. External tooling brings dependencies on 7z.exe / PS scripts or whatever it is, making some inconsistency across the builds and build systems. Either have a local build script and tooling around, or have additional configuration on build server. And never ending "it works on my machine" :)

Again, totally agree on never-used case and just adding some context around Zip usage.

**Comment by daveaglick** [comment]
Can we please try not to call ideas presented here "dumb". Lots of people have lots of different use cases and one of the fundamental reasons for this project to exist (as opposed to other static generators) is that it's flexible enough to handle those other use cases. It's fine to argue that a feature proposal is unneeded, redundant, or a lower priority, but I don't want to curtail idea generation for fear of having an idea be called "dumb".

Not to mention, I can actually see several good use cases for a Zip module. Not just the zipping up of content after generation (which would be a pretty easy thing to implement - only a couple lines with the right zip library), but more importantly for unzipping and reading zipped content. I could three modules being part of a zip module library:
- `Zip`, which zips up content either in the output folder or somewhere else
- `Unzip`, which unzips a zip either to the output folder or somewhere else
- `ReadZip`, which allows loading the contents of a zip file into documents without actually having to unzip it

**Comment by avishnyakov** [comment]
Indeed. C'mon guys, all good. Discussing and exploring ideas from different angles, nothing wrong with it :)

**Comment by daveaglick** [comment]
Now that the file provider mechanism is complete, this will be better addressed by #255

**Comment by avishnyakov** [comment]
:+1:
